src/components/Carousel.vue

```diff
@@ -0,0 +1,276 @@
+        const radius = (largestMinDistance / 2) / Math.sin(centerAngle);
+        this.radius.tweenTo(radius, tween ? this.animationDuration : 0);
+        (this.$el as HTMLElement).style.minHeight = `${largestHeight}px`;
+        this._rerender();
+    }
+
+    @Watch('effectiveSelected')
+    @Watch('disabled')
+    private _updateRotations(newWatcherValueOrTween = true, previousWatcherValue?) {
+        const tween = typeof newWatcherValueOrTween === 'boolean' && typeof previousWatcherValue === 'undefined'
+            ? newWatcherValueOrTween // specified whether to tween
+            : true; // did not specify whether to tween or method was called as a watcher (default to true)
+        // clean up removed entries
+        for (const entry of this.rotations.keys()) {
+            if (this.entries.indexOf(entry) !== -1) continue;
+            this.rotations.delete(entry);
+        }
+        // update rotations
+        for (const entry of this.entries) {
+            const rotation = this.rotations.get(entry) || new Tweenable();
+            const tweenTime = tween ? this.animationDuration : 0;
+            rotation.tweenTo(this._calculateTargetRotation(entry, rotation.currentValue), tweenTime);
+            this.rotations.set(entry, rotation);
+        }
+        this._rerender();
+    }
+
+    private _calculateTargetRotation(entry, currentRotation): number { // rotation in radians
+        if (this.disabled && entry !== this.effectiveSelected) {
+            // hide not selected entries at other end of circle
+            return currentRotation + this._calculateRotationInClosestDirection(currentRotation, Math.PI);
+        }
+        const stepSize = 2 * Math.PI / this._totalPositionCount;
+        const entryIndex = this.entries.indexOf(entry);
+        const selectedIndex = this.entries.indexOf(this.effectiveSelected);
+        let offset = entryIndex - selectedIndex;
+        if (this._hasDummyPosition && offset > this._totalPositionCount / 2) {
+            // skip dummy position
+            offset += 1;
+        }
+        return currentRotation + this._calculateRotationInClosestDirection(currentRotation, offset * stepSize);
+    }
+
+    @Watch('hideBackgroundEntries')
+    private _rerender() {
+        if (this.requestAnimationFrameId !== null) return;
+        this.requestAnimationFrameId = requestAnimationFrame(() => {
+            const zCoordinatesForEntries: Array<[string, number]> = [];
+            let finished = this.radius.finished;
+            for (const [entry, rotation] of this.rotations) {
+                const currentRotation = rotation.currentValue;
+                const currentRadius = this.radius.currentValue;
+                const x = Math.sin(currentRotation) * currentRadius;
+                const z = Math.cos(currentRotation) * currentRadius - currentRadius;
+                const [ el ] = this.$refs[entry];
+                el.style.transform = `translate3d(calc(${x}px - 50%),-50%,${z}px)`;
```

Comment:
As discussed, the front-entry is currently blurry because `x` does not go to complete `0`.

It would maybe also be nice to achieve a `translate3d(0, 0, 0)` for the front entry.

Reply:
Actually, on closer examination, it's blurry because the Y-axis get a -50%, and the height of the panel is uneven, thus lifting it up by a decimal pixel number. Firefox does not display decimal pixels, so it's sharp there, but Chrome does.

Reply:
See https://github.com/nimiq/vue-components/issues/69